src/ServicePulse.Host/app/js/views/event_log_items/eventLogItems.js

```diff
@@ -27,7 +27,11 @@
                     $location.path('/endpoints');
                     break;
                 case 'MessageFailures':
-                    $location.path('/failed-messages/groups');
+                    var newlocation = '/failed-messages/groups';
+                    if (eventLogItem.related_to && eventLogItem.related_to[0].search('message') > 0) {
+                        newlocation = '/failed-messages' + eventLogItem.related_to[0];
```

Comment:
Does `related_to` already include a leading `/`?

Reply:
Answering myself: Yes. Yes it does.

Reply:
yeah it does, what it has is /message/<guid>

I don't like the fact that we assume it will had 0 index in the array. And that it contains such string that is why I wrapped it in the if, so that we fall back to the good old: redirect to the group view